specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json

```diff
@@ -532,6 +561,129 @@
+      "x-ms-discriminator-value": "Unknown",
+      "properties": {
+        "version": {
+          "description": "Defines version of the rule set.",
+          "type": "string"
+        },
+        "ruleSetType": {
+          "description": "RuleSetType - DefaultRuleSet or CRS RuleSet.",
+          "type": "string"
+        }
+      },
+      "required": [
+        "ruleSetType"
+      ],
+      "discriminator": "ruleSetType"
+    },
+    "CrsRuleSet": {
+      "type": "object",
+      "description": "Describes azure managed provider.",
+      "x-ms-discriminator-value": "CrsRuleSet",
+      "properties": {
+        "ruleSetConfiguration": {
+          "$ref": "#/definitions/CrsRuleSetConfiguration",
+          "description": "OWASP rule set configuration."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/ManagedRuleSet"
+        }
+      ]
+    },
+    "CrsRuleSetConfiguration": {
+      "properties": {
+        "disabledRuleGroups": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CrsDisabledRuleGroupEntry"
+          },
+          "description": "The disabled rule groups."
+        }
+      },
+      "description": "OWASP rule set configuration."
+    },
+    "CrsDisabledRuleGroupEntry": {
+      "properties": {
+        "ruleGroupName": {
+          "type": "string",
+          "description": "The name of the rule group that will be disabled."
+        },
+        "rules": {
+          "type": "array",
+          "items": {
+            "type": "integer",
+            "format": "int32",
+            "x-nullable": false
+          },
+          "description": "The list of rules that will be disabled. If null, all rules of the rule group will be disabled."
+        }
+      },
+      "required": [
+        "ruleGroupName"
+      ],
+      "description": "Allows to disable rules within a rule group or an entire rule group."
+    },
+    "CrsExclusions": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/CrsExclusionEntry"
+      },
+      "description": "The exclusion list."
+    },
+    "CrsExclusionEntry": {
+      "properties": {
+        "matchVariable": {
+          "type": "string",
+          "description": "The variable to be excluded."
+        },
+        "selectorMatchOperator": {
+          "type": "string",
+          "description": "When matchVariable is a collection, operate on the selector to specify which elements in the collection this exclusion applies to."
+        },
+        "selector": {
+          "type": "string",
+          "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
+        }
+      },
+      "required": [
+        "matchVariable"
+      ],
+      "description": "Allow to exclude some variable satisfy the condition for the WAF check."
+    }
+  },
+  "parameters": {
```

Comment:
Please don't duplicate existing definitions, use `SubscriptionIdParameter` and `ApiVersionParameter` from `network.json`,

Reply:
Yeah. Makes sense!